.github/workflows/release-azure-pipelines.yml

```diff
@@ -93,40 +93,268 @@ extends:
+                    "Parameters": {}
+                 },
+                 {
+                  "KeyCode": "$(SIGNING_KEY_CODE_MAC)",
+                  "OperationCode": "SigntoolVerify",
+                  "ToolName": "sign",
+                  "ToolVersion": "1.0",
+                  "Parameters": {}
+                 }
+                ]
+              SessionTimeout: '60'
+              MaxConcurrency: '50'
+              MaxRetryAttempts: '5'
+              PendingAnalysisWaitTimeoutMinutes: '5'
+          - task: ExtractFiles@1
+            displayName: Extract signed artifacts osx
+            condition: startsWith('${{ config.runtime }}', 'osx')
+            inputs:
+              archiveFilePatterns: $(Build.ArtifactStagingDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}.zip
+              destinationFolder: $(Build.ArtifactStagingDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}
+              cleanDestinationFolder: true
+              overwriteExistingFiles: true
+          # We rename the signed artifacts to avoid conflicts with the unsigned pipeline artifacts from the previous stage.
+          - task: PowerShell@2
+            displayName: Rename signed artifacts
+            inputs:
+              workingDirectory: $(Build.ArtifactStagingDirectory)
+              targetType: 'inline'
+              script: |
+                mv "azureauth-${{ parameters.version }}-${{ config.runtime }}" "azureauth-${{ parameters.version }}-${{ config.runtime }}-signed"
 
-            - task: DotNetCoreCLI@2
-              displayName: Install dependencies
-              inputs:
-                command: restore
-                feedsToUse: select
-                vstsFeed: $(vstsFeedId)
-                includeNuGetOrg: false
-                arguments: --runtime ${{ config.runtime }}
+    # Currently we package artifacts into the most commonly accessible archive format for their respective platforms.
+    - stage: package
+      displayName: Package
+      dependsOn: sign
+      jobs:
+        - job: package
+          displayName: Package
+          pool:
+            name: Azure-Pipelines-1ESPT-ExDShared
+            image: ubuntu-latest
+            os: linux
+          templateContext:
+            inputs:
+            - ${{ each config in parameters.buildConfigs }}:
+              - input: pipelineArtifact
+                artifactName: azureauth-${{ parameters.version }}-${{ config.runtime }}-signed
+                targetPath: $(Build.ArtifactStagingDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}-signed
+            outputs:
+            - ${{ each config in parameters.buildConfigs }}:
+              - output: pipelineArtifact
+                artifactName: azureauth-${{ parameters.version }}-${{ config.runtime }}.${{ config.archiveExt }}
+                targetPath: $(Build.SourcesDirectory)/azureauth-${{ parameters.version }}-${{ config.runtime }}.${{ config.archiveExt }}

```

Comment:
If the `targetPath` here is used to place the final artifacts, you can try using `Build.BinariesDirectory` instead as `$(Build.SourcesDirectory)` should ideally be reserved for the source code files. See https://learn.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml#build-variables-devops-services for more information on pre-defined pipeline variables.

Reply:
So the `targetPath` here refers to from which path should artifacts be uploaded at the end of stage. Since the packaging step compresses and puts it in sources directory, I have provided that path (as compressed file is only available at that path to upload it). I tried searching for specifying the target directory for compressed file itself so the pipeline can pick up from there to upload it but I was not able to find that location (It compresses and puts it in the same place from where it compressed). I can have an additional task to copy the compressed files to `BinariesDirectory`, but I am not sure it will add any value (as this is just the location to pick up from)

Reply:
SBOM (when switching to prod pipeline) requires output to be a folder and not file (like we do zip/tar.gz here...sigh). I am going to change this so that we have `packaged` directory similarly to `signed` one. I will put this change in #394 PR

Reply:
Oh then I don't think it's worth doing an additional copy. It makes sense as the final compressed signed files are places in the same location of the original file. It would have been great if the signing task provided an option to specify the output file location.